services/ui_backend_service/api/autocomplete.py

```diff
@@ -0,0 +1,110 @@
+from services.utils import handle_exceptions, web_response
+import asyncio
+import threading
+
+
+class AutoCompleteApi(object):
+    # Cache tags so we don't have to request DB everytime
+    tags = []
+
+    def __init__(self, app, db):
+        self.db = db
+        # Cached resources
+        app.router.add_route("GET", "/autocomplete/tags", self.get_all_tags)
+        # Non-cached resources
+        app.router.add_route("GET", "/autocomplete/flows", self.get_all_flows)
+        app.router.add_route("GET", "/autocomplete/flows/{flow_id}/runs", self.get_runs_for_flow)
+        app.router.add_route("GET", "/autocomplete/flows/{flow_id}/runs/{run_id}/steps", self.get_steps_for_run)
+        self._async_table = self.db.run_table_postgres
+        self.loop = asyncio.get_event_loop()
+        self.loop.create_task(self.setup_tags())
+
+    @handle_exceptions
+    async def setup_tags(self):
+        db_response = await self._async_table.get_tags()
+
+        if db_response.response_code == 200:
+            self.tags = db_response.body
+        # Check tags again in 5 minutes
+        self.loop.call_later(300, lambda x: self.loop.create_task(self.setup_tags()), self)
+
+    @handle_exceptions
+    async def get_all_tags(self, request):
+        """
+        ---
+        description: Get all available tags
+        tags:
+        - Autocomplete
+        produces:
+        - application/json
+        responses:
+            "200":
+                description: Returns string list of all tags
+                type: array
+                items:
+                    type: string
+        """
+        return web_response(200, self.tags)
+
+    @handle_exceptions
+    async def get_all_flows(self, request):
+        """
+        ---
+        description: Get all flow id's as a list
+        tags:
+        - Autocomplete
+        produces:
+        - application/json
+        responses:
+            "200":
+                description: Returns string list of all flow id's
+                type: array
+                items:
+                    type: string
+        """
+        db_response = await self._async_table.get_field_from(field="flow_id", table="flows_v3")
```

Comment:
table names should be pulled directly from the db class instead of hardcoding, as these are subject to change whenever. So for this case rather use `table=self._async_table.table_name`